frontend/src/layout/navigation/AsyncActionWarning.tsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react'
+import { kea, useActions, useValues } from 'kea'
+import { Alert } from 'antd'
+import { InfoCircleOutlined } from '@ant-design/icons'
+import { userLogic } from 'scenes/userLogic'
+
+const asyncWarningLogic = kea({
+    actions: {
+        dismissWarning: true,
+    },
+    reducers: {
+        warningDismissed: [
+            false,
+            { persist: true }, // dismiss forever if dismissed
+            {
+                dismissWarning: () => true,
+            },
+        ],
+    },
+})
+
+export function AsyncActionWarning(): JSX.Element | null {
+    const { user } = useValues(userLogic)
+    const { warningDismissed } = useValues(asyncWarningLogic)
+    const { dismissWarning } = useActions(asyncWarningLogic)
+
+    if (warningDismissed || !user?.is_async_event_action_mapping_enabled) {
+        return null
+    }
+    return (
+        <>
+            <Alert
+                type={'warning'}
+                message={'Webhooks and actions delayed up to 5 minutes'}
+                className="demo-warning"
+                description={
+                    <>Due to temporary limitations, webhooks and actions might be delayed by up to 5 minutes.</>
```

Comment:
I don't think this will be temporary?

Reply:
Until we move action event mapping to the plugin server, as discussed here: https://github.com/PostHog/plugin-server/issues/235

Reply:
But will action mapping in the plugin server by itself make things better? 🤔 Postgres is the bottleneck in this case and action mapping on the fly will (did?) in fact hammer it even more, because every event is a whole new request. I think what may need optimization is action mapping queries/data layout itself. Otherwise action matching without significant delays seems to be hard to pull off. Feel free to correct me

Reply:
Action event mapping in the plugin server wouldn't make extra queries, that's the thing :).

Except if a) needing to fetch cohorts to see if a person belongs to them due to a filter, or b) fetching a person to match person properties in a filter. The person is more often than not already fetched, and we could inline adding the cohorts with a subquery if needed.